module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -98,6 +100,20 @@ public function holdAction()
             return $this->redirectToRecord('#top');
         }
 
+        // Attach holdings data from requested item for template use
+        $requestedItemId = $this->params()->fromPost('item_id') ?: $this->params()->fromQuery('item_id');
+        if ($requestedItemId) {
+            $holdings = $catalog->getHolding($gatheredDetails['id'], $patron);
+            $requestedItem = null;
+            foreach ($holdings['holdings'] as $item) {
+                if ($item['item_id'] === $requestedItemId) {
+                    $requestedItem = $item;
+                    break;
+                }
+            }
+            $gatheredDetails['requestedItem'] = $requestedItem;
+        }
```

Comment:
This is really a matter of opinion, so feel free to disregard if you prefer it as-is, but maybe we could cut out an extra assignment by just using the array element directly:

```suggestion
        // Attach holdings data from requested item for template use
        $requestedItemId = $this->params()->fromPost('item_id') ?: $this->params()->fromQuery('item_id');
        if ($requestedItemId) {
            $holdings = $catalog->getHolding($gatheredDetails['id'], $patron);
            $gatheredDetails['requestedItem'] = null;
            foreach ($holdings['holdings'] as $item) {
                if ($item['item_id'] === $requestedItemId) {
                    $gatheredDetails['requestedItem'] = $item;
                    break;
                }
            }
        }
```